File: 2006-count-number-of-pairs-with-absolute-difference-k/2006-count-number-of-pairs-with-absolute-difference-k.py
from collections import defaultdict
class Solution:
    def countKDifference(self, nums: List[int], k: int) -> int:
        mp = defaultdict(lambda : 0)
        count = 0
        for i in nums:
            if(i + k in mp):
                count += mp[i + k]
            if(i - k in mp):
                count += mp[i-k]
            mp[i] += 1
        return count
            
        

File: 2006-count-number-of-pairs-with-absolute-difference-k/2006-count-number-of-pairs-with-absolute-difference-k.java
class Solution {
    public int countKDifference(int[] nums, int k) {
        int ans = 0;
        
        Map<Integer,Integer> m = new HashMap<>();
        for(int i=0;i<nums.length;i++){
            
            if(m.containsKey(nums[i]+k))
                ans+=m.get(nums[i]+k);
                
            if(m.containsKey(nums[i]-k))    
                ans+=m.get(nums[i]-k);
               
            m.put(nums[i],m.getOrDefault(nums[i],0) +1);
        }
        
         
        return ans;
        
    }
}